libsolidity/parsing/Parser.cpp

```diff
@@ -1056,6 +1056,28 @@ ASTPointer<Expression> Parser::parsePrimaryExpression()
 		expectToken(Token::RParen);
 		return nodeFactory.createNode<TupleExpression>(components);
 	}
+	case Token::LBrack:
+	{
+		// Inline array expression
+		// Special cases: [] is empty tuple type, (x) is not a real tuple, (x,) is one-dimensional tuple
```

Comment:
hmmm...but why not just make that an empty cell then? Could you not just make it so you're essentially assigning an array of 2 with one slot filled and another just empty? Seems it could have some uses.